Formula/azure-cli.rb

```diff
@@ -16,26 +13,493 @@ class AzureCli < Formula
+  end
+
+  resource "Whoosh" do
+    url "https://files.pythonhosted.org/packages/25/2b/6beed2107b148edc1321da0d489afc4617b9ed317ef7b72d4993cad9b684/Whoosh-2.7.4.tar.gz"
+    sha256 "7ca5633dbfa9e0e0fa400d3151a8a0c4bec53bd2ecedc0a67705b17565c31a83"
+  end
+
+  resource "xmltodict" do
+    url "https://files.pythonhosted.org/packages/57/17/a6acddc5f5993ea6eaf792b2e6c3be55e3e11f3b85206c818572585f61e1/xmltodict-0.11.0.tar.gz"
+    sha256 "8f8d7d40aa28d83f4109a7e8aa86e67a4df202d9538be40c0cb1d70da527b0df"
+  end
+
+  # Patch to support 'az extension add' due to https://github.com/Azure/azure-cli/issues/4428
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/e600681/azure-cli/bug-4428.patch"
+    sha256 "e834613b7d983fca591e2202b6dbe1feab106fdb84edb20ca475828b32d88e61"
+  end
 
   def install
-    rm_rf "bin/windows"
+    xy = Language::Python.major_minor_version "python3"
+    site_packages = libexec/"lib/python#{xy}/site-packages"
+    ENV.prepend_create_path "PYTHONPATH", site_packages
+    ENV.prepend "LDFLAGS", "-L#{Formula["openssl"].opt_lib}"
+    ENV.prepend "CFLAGS", "-I#{Formula["openssl"].opt_include}"
+    ENV.prepend "CPPFLAGS", "-I#{Formula["openssl"].opt_include}"
+
+    # Get the CLI components we'll install
+    components = [
+      buildpath/"src/azure-cli",
+      buildpath/"src/azure-cli-core",
+      buildpath/"src/azure-cli-nspkg",
+      buildpath/"src/azure-cli-command_modules-nspkg",
+    ]
+    components += Pathname.glob(buildpath/"src/command_modules/azure-cli-*/")
 
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
-    (bash_completion/"azure").write Utils.popen_read("#{bin}/azure --completion")
+    # Install dependencies
+    # note: Even if in 'resources', don't include 'futures' as not needed for Python3
+    # and causes import errors. See https://github.com/agronholm/pythonfutures/issues/41
+    deps = resources.map(&:name).to_set - ["futures"]
+    deps.each do |r|
+      resource(r).stage do
+        system "python3", *Language::Python.setup_install_args(libexec)
+      end
+    end
+
+    # Install CLI
+    components.each do |item|
+      cd item do
+        system "python3", *Language::Python.setup_install_args(libexec)
+      end
+    end
+
+    # This replaces the `import pkg_resources` namespace imports from upstream
+    # with empty string as the import is slow and not needed in this environment.
+    File.open(site_packages/"azure/__init__.py", "w") {}
+    File.open(site_packages/"azure/cli/__init__.py", "w") {}
+    File.open(site_packages/"azure/cli/command_modules/__init__.py", "w") {}
+    File.open(site_packages/"azure/mgmt/__init__.py", "w") {}
+
+    az_exec = <<-EOS.undent
+      #!/usr/bin/env bash
+      export PYTHONPATH="#{ENV["PYTHONPATH"]}"
+      python#{xy} -m azure.cli \"$@\"
+    EOS
+    (bin/"az").write az_exec
```

Comment:
```
    (bin/"az").write <<-EOS.undent
      #!/usr/bin/env bash
      export PYTHONPATH="#{ENV["PYTHONPATH"]}"
      python#{xy} -m azure.cli \"$@\"
    EOS
```

That said, it would be nice if this could use the `write_env_script` DSL instead.

Reply:
okay changed this...